app/grants/models.py

```diff
@@ -474,22 +469,24 @@ def get_hash_from_web3(self):
             args['nonce'],
             ).call()
 
-
-    def successful_contribution(self, kwargs):
+    def successful_contribution(self, tx_id):
         """Create a contribution object."""
         from marketing.mails import successful_contribution
         self.last_contribution_date = timezone.now()
-        self.next_contribution_date = timezone.now() + timezone.timedelta(seconds=self.real_period_seconds)
+        self.next_contribution_date = timezone.now() + timezone.timedelta(seconds=int(self.real_period_seconds))
         self.save()
+        # TODO: add gasPrice and nonce
         contribution_kwargs = {
-            'tx_id': kwargs.tx_id,
-            'gas_price': kwargs.gas_price,
-            'nonce': kwargs.nonce,
+            'tx_id': tx_id,
             'subscription': self
         }
         contribution = Contribution.objects.create(**contribution_kwargs)
         grant = self.grant
-        grant.amount_received = (grant.amount_received + convert_amount(self.amount_per_period, self.token_symbol, "USDT", timezone.now()))
+        grant.amount_received = (
```

Comment:
i wonder if its worth making `amount_received` a JSON blob that has a keyvalue pair of all currencies being added to the grant

i.e.
```
{ 'WETH': 0.01, 'DAI': 12}
```

Reply:
i'm not sure what the scope of this PR is so i can't say.

it doesnt seem like a blocking thing to me though if the scope of this is to cleanup the subminer